pkg/build/builder/cmd/builder.go

```diff
@@ -106,19 +110,21 @@ func setupSourceSecret(sourceSecretName string, scmAuths []scmauth.SCMAuth) erro
 	}
 
 	if len(scmAuthsPresent) == 0 {
-		return fmt.Errorf("no auth handler was found for the provided secret %q",
+		return nil, fmt.Errorf("no auth handler was found for the provided secret %q",
 			sourceSecretName)
 	}
 
+	url := &url.URL{}
+
 	for name, auth := range scmAuthsPresent {
 		glog.V(3).Infof("Setting up SCMAuth %q", name)
-		if err := auth.Setup(sourceSecretDir); err != nil {
+		if url, err = auth.Setup(sourceSecretDir); err != nil {
```

Comment:
this is hard to follow... the last auth will win, even if an earlier one set to non-nil